### PR TITLE
Add win32 support to python release test script

### PR DESCRIPTION
**Pull request by tolbrino**
This PR add support for running the python release tests on Windows. Differences to running on Linux/macOS are:

- Use of `.zip` file instead of `.tar.gz`
- Use of Python3 instead of Python2, because a Python2 setup wasn't possible
- Use of OS package manager for some Python dependencies, while the rest is installed via pip outside of any virtualenv.
- Simulating user behaviour on Windows by using node `console` instead of `start`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ST_CT_FLAGS = --logdir system_test/logs
 ST_CT_DIR = --dir system_test/common
 ST_CT_LOCALDIR = --dir system_test/only_local
 
-SWAGGER_CODEGEN_CLI_V = 2.3.1
+SWAGGER_CODEGEN_CLI_V = 2.4.4
```
Comment (lucafavatella) [inline Makefile]: I do not expect changes in code included in node but only in Python HTTP client. Ok.

File: scripts/windows/msys2_prepare.bat [hunk 2]
```diff
@@ -13,13 +16,16 @@ rem Set required vars defaults
 IF "%ERTS_VERSION%"=="" SET "ERTS_VERSION=9.3"
 IF "%OTP_VERSION%"=="" SET "OTP_VERSION=20.3"
 IF "%WIN_OTP_PATH%"=="" SET "WIN_OTP_PATH=C:\Program Files\erl"
+IF "%WIN_JDK_BASEPATH%"=="" SET "WIN_JDK_BASEPATH=C:\Program Files\Java"
```
Comment (lucafavatella) [inline scripts/windows/msys2_prepare.bat]: Why is JDK needed? This is run by user - see docs/build-windows.md
If not needed by user, it looks like this shall be separate script for CI only?

Comment (lucafavatella) [inline scripts/windows/msys2_prepare.bat]: Shall these new params documented for user? See docs/build-windows.md

File: scripts/windows/msys2_prepare.bat [hunk 3]
```diff
@@ -30,18 +36,15 @@ make ^
...
 mingw-w64-x86_64-binutils ^
 mingw-w64-x86_64-gcc ^
 mingw-w64-x86_64-libsodium ^
 mingw-w64-x86_64-nsis ^
 mingw-w64-x86_64-ntldd-git  ^
-mingw-w64-x86_64-python3 ^
-mingw-w64-x86_64-python3-pip ^
 mingw-w64-x86_64-yasm ^
 patch ^
+unzip ^
```
Comment (lucafavatella) [inline scripts/windows/msys2_prepare.bat]: Needed only for CI? Or also for user?

File: py/tests/release.py [hunk 4]
```diff
@@ -180,33 +174,77 @@ def node_is_online(api):
...
+    assert os.path.isfile(binary), "Not a file: " + binary
+    assert os.access(binary, os.X_OK), "Can't access file: " + binary
+    print("Stopping " + binary)
+    os.chdir(temp_dir)
+    if sys.platform == "win32":
+        # We need to find the PID listening on the right port
+        listen_address = SETUP[node]["listen_address"]
+        result_lines = subprocess.check_output("netstat -ano -p TCP").decode("utf-8").split("\r\n")
+        pid = functools.reduce(lambda acc, r: find_pid(r, listen_address, acc), result_lines, "")
+        os.kill(int(pid), signal.SIGTERM)
```
Comment (lucafavatella) [inline py/tests/release.py]: So this ^^^ is the way the user is meant to stop node on Window? 🤔

File: py/tests/release.py [hunk 5]
```diff
@@ -180,33 +174,77 @@ def node_is_online(api):
...
+    assert os.path.isfile(binary), "Not a file: " + binary
+    assert os.access(binary, os.X_OK), "Can't access file: " + binary
     print("Starting " + binary)
     os.chdir(temp_dir)
-    os.system('ERL_FLAGS="-config `pwd`/p.config" bin/aeternity start')
+    erl_flags = " ".join(SETUP[node]["config"])
+    if sys.platform == "win32":
+        sub_env = os.environ
+        sub_env["ERL_FLAGS"] = erl_flags
+        subprocess.Popen([binary, "console"], env=sub_env)
```
Comment (lucafavatella) [inline py/tests/release.py]: Is there something closer to how user would pass flags?

File: py/tests/release.py (hunk 4 above)
Comment (tolbrino) [reply py/tests/release.py]: Particularly on Windows the user works via the `werl.exe` binary with the node. This emulates a terminal as a proper window. The user can just close that window to kill the node. However, to emulate that the release test has to jump through the hoops and work via the PID.

File: py/tests/release.py (hunk 5 above)
Comment (tolbrino) [reply py/tests/release.py]: Using `ERL_FLAGS` is the way any user would do that. The difference to the previous approach is that one can't pass multiple configuration file arguments on Windows, which is why I had to refactor it.

File: scripts/windows/msys2_prepare.bat (hunk 2 above)
Comment (tolbrino) [reply scripts/windows/msys2_prepare.bat]: Yes should be.

Comment (tolbrino) [reply scripts/windows/msys2_prepare.bat]: The files under `scripts/windows` are meant for users wanting to build the node. Therefore, the JDK is required to generate the swagger test client. So far there wasn't a differentiation between such a user and a developer using Windows.

File: scripts/windows/msys2_prepare.bat (hunk 3 above)
Comment (tolbrino) [reply scripts/windows/msys2_prepare.bat]: Needed for the release testing.

File: apps/aehttp/priv/swagger.json [hunk 6]
```diff
@@ -106,7 +106,13 @@
           "200" : {
             "description" : "Successful operation",
             "schema" : {
-              "$ref" : "#/definitions/inline_response_200"
```
Comment (lucafavatella) [inline apps/aehttp/priv/swagger.json]: No refs. So schema shall be more explicit. Ok.

File: apps/aehttp/priv/swagger.json [hunk 7]
```diff
@@ -2212,7 +2238,9 @@
       "items" : {
         "type" : "integer",
         "format" : "int32"
-      }
+      },
+      "minItems" : 42,
```
Comment (lucafavatella) [inline apps/aehttp/priv/swagger.json]: More precise. Ok.

File: apps/aehttp/priv/swagger.json [hunk 8]
```diff
@@ -2267,15 +2295,10 @@
         }
       },
       "example" : {
-        "prev_hash" : null,
```
Comment (lucafavatella) [inline apps/aehttp/priv/swagger.json]: Less noise. Ok.

File: py/requirements.txt [hunk 9]
```diff
@@ -1,3 +1,10 @@
+##
+## When updating these dependencies you must ensure that the relevant entries in the following files are kept up-to-date.
+## as well. This is required due to the more complicated python setup on Windows/msys2. Look for the tag WINDOWS_PYTHON.
+##
+## - scripts/windows/msys2_prepare.py
+## - py/requirements-win32.txt
+##
```
Comment (lucafavatella) [inline py/requirements.txt]: ❤️

File: appveyor.yml [hunk 10]
```diff
@@ -25,12 +25,12 @@ environment:
       OTP_VERSION: 20.1
       ERTS_VERSION: 9.1
       BUILD_STEP: build
-      TEST_STEP: ct
+      TEST_STEPS: ct,release
```
Comment (lucafavatella) [inline appveyor.yml]: I am a bit surprised that these are note rendered as 4 jobs:
* OTP 20.1 ct - artefacts being CT logs and node logs
* OTP 20.3 ct - artefacts being CT logs and node logs
* OTP 20.1 release - artefacts packages, only if smoke test passes
* OTP 20.3 release - artefacts packages, only if smoke test passes

If too slow, the 20.1 jobs may be deleted.

If too complex, they can be kept as is.

Comment (tolbrino) [reply appveyor.yml]: On AppVeyor CI it is not possible to share build artifacts between jobs. Therefore, splitting the test steps into 2 job would result in an additional build/package cycle which adds quite some time. That is why I condensed it into a single job.

File: scripts/windows/msys2_prepare.bat [hunk 11]
```diff
@@ -93,6 +86,13 @@ rem Remove breaking tools
 
 %BASH_BIN% -lc "%PACMAN_RM% %PACMAN_PACKAGES_REMOVE% || true"
 
+@echo Current time: %time%
+rem Upgrade the MSYS2 platform
+
+%BASH_BIN% -lc "%PACMAN% -y pacman"
```
Comment (lucafavatella) [inline scripts/windows/msys2_prepare.bat]: Is this installing `pacman`?
If so, how can `packman` be present earlier?

**Comment by lucafavatella** [comment]
I understand WIP so I took the liberty to label it as so.

**Comment by tolbrino** [comment]
@lucafavatella @dincho This PR is finished, ready for final review.

**Comment by dincho** [comment]
@noandrea could you please confirm that there is no BC breaking changes in swagger schemas ?

**Comment by tolbrino** [comment]
> @tolbrino Could you please confirm that if smoke test fails then CI marks job as failed.

Here is an example of the test failing and the CI marking it as such: https://ci.appveyor.com/project/aeternity/aeternity/builds/23804790/job/aj79oj67doq6wkqf

**Pull request merged**